src/inmanta/ast/attribute.py

```diff
@@ -117,17 +108,7 @@ def validate(self, value: object) -> None:
         self.type.validate(value)
 
     def get_new_result_variable(self, instance: "Instance", queue: QueueScheduler) -> ResultVariable:
-        out: ResultVariable["Instance"]
-
-        if self.is_optional():
-            # be a 0-1 relation
-            self.end = None
-            self.low = 0
-            self.high = 1
-            out = DeprecatedOptionVariable(self, instance, queue)
-        else:
-            out = ResultVariable()
-
+        out: ResultVariable["Instance"] = ResultVariable()
```

Comment:
I don't think this is typed correctly (as in, it has been typed incorrectly for the past four years). Could you fix it and run `mypy-diff`? I think `ResultVariable[object]` is probably the best we can do.